### ISSUE TITLE
Rounding causes total consumption to never be greater than 0.00 kWH

### ISSUE THREAD

**Issue by toofishes** [ebaauw/homebridge-ups#8]
The fix merged for #5 and #6 seems to have caused a new issue- total consumption now always shows as 0.00 kWh, even if the UPS is reporting usage. This also affects the Eve-computed Total Cost value.

<img src="https://user-images.githubusercontent.com/265817/198885476-d3ead63c-5654-4181-bc3b-247758894d51.jpg" width=250>

I think (but haven't totally proven it) that it is related to the rounding done in these two places:
* https://github.com/ebaauw/homebridge-lib/blob/c4fa195194d03bca86bb6595b699c1c0573e7c5d/lib/ServiceDelegate/History/Power.js#L81
* https://github.com/ebaauw/homebridge-lib/blob/c4fa195194d03bca86bb6595b699c1c0573e7c5d/lib/ServiceDelegate/History/Power.js#L114

Since this value is published as a `FLOAT` characteristic, I think it could either omit rounding completely, or replace:
```js
this._totalConsumption += Math.round(delta / 3600000.0) // kWh
```
with
```js
this._totalConsumption += Math.round(delta / 3600.0) / 1000.0 // kWh
```
To round to three decimal places (e.g. Wh, but expressed as kWh).

**Comment by ebaauw** [comment]
I think you're right.  Damn, this proves harder than I expected.  Normally, smart plugs report _Total Consumption_ and I simply expose that.  This is the first time I actually try and compute it from _Current Consumption_.

Double checking my Eve Energy smart plugs, Eve seems to show _Total Consumption_ with a  0.01 kWh resolution, and _Current Consumption_ with a 1W resolution.  The history seems to have entries with a 0.01Wh resolution, but that's deceiving.  It shows values in Wh, but the values themselves are integer, in 0.1 W * 10 min.  So it divides those by 6 (10 min/h) to get 0.1 Wh.   I see I round those values to W, and then multiply by 10.

I think, it makes sense only to round values when sending these to HomeKit and keep the internal accumulation in full precision.  Let me see if I can get it right this time...

**Comment by ebaauw** [comment]
Ran the changes for a day in test; it looks to be correct now.  Published v1.0.5.

**Comment by toofishes** [comment]
I've been running 1.0.5 for about 24 hours now, and I am also seeing total consumption data that looks good. Thanks for fixing!
